### ISSUE TITLE
Restore job-based granularity

### ISSUE THREAD

**Issue by csadorf** [glotzerlab/signac-flow#37]
**[Original report](https://bitbucket.org/glotzer/signac-flow/issue/37) by Jens Glaser (Bitbucket: [jens_glaser](https://bitbucket.org/jens_glaser), GitHub: [jglaser](https://github.com/jglaser)).**

----------------------------------------

I have to admit that I don't quite understand the submission interface. Currently it maps operations onto real cluster jobs. However, that may be a finer level of granularity than one really wants if the operations are quick, but the job scheduler throughput is slow. One may end up waiting for the next cluster slot to free up between two trivial tasks. I don't see an option to generate scripts that basically contain

```
python project.py run -j <my_job_id>
```

In other words, bundle all job-specific tasks into one single cluster job.

Alternatively, the execution graph could be traversed, as proposed for issue #35, and only the last operation would be issued, automatically fulfilling previous dependencies.

**Comment by csadorf** [comment]
This seems to be related to issue #30 .

**Comment by vyasr** [comment]
This issue should be solved by #114, and is also closely related to #35 if we want to make this even easier.

**Comment by vyasr** [comment]
The original question raised in this issue -- the ability to submit multiple operations within a single cluster job -- is resolved by the addition of groups (#114). Addressing #35 will make it even easier to ensure that all preconditions are satisfied, but that is a separate issue so this one can be closed.
